docs/powerpoint/tagging-presentations-slides-shapes.md

```diff
@@ -0,0 +1,162 @@
+
+2. Create a method to get the index of the selected slide. The following is an example. Note about this code:
+
+    - It uses the [Office.context.document.getSelectedDataAsync](/javascript/api/office/office.document#getSelectedDataAsync_coercionType__callback_) method of the Common JavaScript APIs.
+    - The call to `getSelectedDataAsync` is embedded in a Promise-returning function. For more information about why and how to do this, see [Wrap Common APIs in promise-returning functions](../develop/asynchronous-programming-in-office-add-ins.md#wrap-common-apis-in-promise-returning-functions).
+    - `getSelectedDataAsync` returns an array because multiple slides can be selected. In this scenario, the user has selected just one, so the code gets the first (0th) slide, which is the only one selected.
+    - The `index` value of the slide is the 1-based value the user sees beside the slide in the thumbnails pane.
+
+    ```javascript
+    function getSelectedSlideIndex() {
+        return new OfficeExtension.Promise<number>(function(resolve, reject) {
+            Office.context.document.getSelectedDataAsync(Office.CoercionType.SlideRange, function(asyncResult) {
+                try {
+                    if (asyncResult.status === Office.AsyncResultStatus.Failed) {
+                        reject(console.error(asyncResult.error.message));
+                    } else {
+                        resolve(asyncResult.value.slides[0].index);
+                    }
+                } 
+                catch (error) {
+                    reject(console.log(error));
+                }
+            });
+        });
+    }
+    ```
+
+3. Create a method to delete slides that are tagged for premium customers. The following is an example. Note about this code:
+
+    - Because the `key` and `value` properties of the tags are going to be read after the `context.sync`, they must be loaded first.
+
+    ```javascript
+    async function deleteSlidesByAudience() {
+      await PowerPoint.run(async function(context) {
+        const slides = context.presentation.slides;
+        slides.load("tags/key, tags/value");
+    
+        await context.sync();
+    
+        for (let i = 0; i < slides.items.length; i++) {
+          let currentSlide = slides.items[i];
+          for (let j = 0; j < currentSlide.tags.items.length; j++) {
+            let currentTag = currentSlide.tags.items[j];
+            if (currentTag.key === "CUSTOMER_TYPE" && currentTag.value === "Premium") {
+              currentSlide.delete();
+            }
+          }
+        }
+    
+        await context.sync();
+      });
+    }
+    ```
+
+## Setting custom metadata on the presentation
+
+Add-ins can also apply tags to the presentation as a whole. This enables you to use tags for document-level metadata similar to how [CustomProperty](/javascript/api/word/word.customproperty)s are used in Word. But unlike Word `CustomProperty`s, the value of a PowerPoint tag can only be of type `string`.
```

Comment:
A suggestion to avoid "CustomPropertys": 
```suggestion
Add-ins can also apply tags to the presentation as a whole. This enables you to use tags for document-level metadata similar to how the [CustomProperty](/javascript/api/word/word.customproperty)class is used in Word. But unlike the Word `CustomProperty` class, the value of a PowerPoint tag can only be of type `string`.
```